e2e/mock-api-v2/src/handlers/authorize.ts

```diff
@@ -0,0 +1,34 @@
+import { Effect } from 'effect';
+import { RouterBuilder } from 'effect-http';
+import { apiSpec } from '../spec';
+import { CookieService } from '../services/Cookie';
+import { toCookieHeader } from '@effect/platform/Cookies';
+import { Fetch } from '../services/fetch';
+
+const authorizeHandler = RouterBuilder.handler(apiSpec, 'DavinciAuthorize', ({ headers, query }) =>
+  Effect.gen(function* () {
+    const { get } = yield* Fetch;
```

Comment:
For what its worth, i'm considering a refactor of how this works. Its a little convoluted to make `Fetch` the all powerful service and make it work perfectly in both test and live environments because of some of the types, and trying to narrow the union of responses appropriately.

What i'm probably going to do is make `fetch` a very basic service.

Then i will make a few `Layers`, specific to the route they in.

This will make it so we have less type complexity in a singular area, and don't have to stress about narrowing down results from `Fetch`. I fear this would become too complex.

So what this would look like is

```
yield* handleAuthorize(headers, query)
```

handleAuthorize would use `Fetch` in it, to "fetch" to the AS, and then we can handle specific validations in here for authorize.

My thought is it would greatly simplify the type narrowing and while `fetch` is probably what we would use in a `bff` setting. I think it just makes sense for the goals of this mock api.

Reply:
I have to think through this a bit though. i haven't fully groked this out in my head.